helix-core/src/test/java/org/apache/helix/integration/rebalancer/CrushRebalancers/TestCrushAutoRebalanceNonRack.java

```diff
@@ -264,6 +266,12 @@ public void testLackEnoughInstances(String rebalanceStrategyName, String rebalan
     System.out.println("TestLackEnoughInstances " + rebalanceStrategyName);
     enablePersistBestPossibleAssignment(_gZkClient, CLUSTER_NAME, true);
 
+    // Drop instance from admin tools and controller sending message to the same instance are
+    // fundamentally async. The race condition can also happen in production.  For now we stabilize
+    // the test by disable controller and re-enable controller to eliminate this race condition as
+    // a workaround. New design is needed to fundamentally resolve the expose issue.
+    _controller.syncStop();
```

Comment:
Stop the controller will work, but it is not what we do in the production.
Why not using the maintenance mode as I suggested originally?

Reply:
That needs a sleep() afterward. Otherwise, it won't work as it take at least one pipeline run cycle there. 

Are we ok with adding a sleep()?

Reply:
Yeah, I get what you said. Let's keep the sync stop here.